app/controllers/api/automate_domains_controller.rb

```diff
@@ -1,5 +1,29 @@
 module Api
   class AutomateDomainsController < BaseController
+    REQUIRED_FIELDS = %w[git_url ref_type ref_name].freeze
+
+    def create_from_git_resource(type, _id, data)
+      assert_all_required_fields_exists(data, type, REQUIRED_FIELDS)
+      raise BadRequestError, 'ref_type must be "branch" or "tag"' unless valid_ref_type?(data)
+
+      api_log_info("Create will be queued for automate domain from #{data["git_url"]} / #{data["ref_name"]}")
```

Comment:
can the git_url include credentials ?

Reply:
@abellotti  I'm sorry but I'm not clear as to how that could be a good approach but am willing to discuss any benefits you may envision.

Reply:
If that's not a supported ref syntax, we're ok then.  If they specify it as such in the inbound request, it will be logged all over since it's not a password type attr, so it's a broader issue than this.